protocol-units/settlement/mcr/contracts/script/MovementDeployer.s.sol

```diff
@@ -0,0 +1,269 @@
+        deployment.staking = address(stakingProxy);
+        deployment.stakingAdmin = _storeAdminDeployment();
+    }
+
+    function _upgradeStaking() internal {
+        console.log("STAKING: upgrading");
+        MovementStaking newStakingImplementation = new MovementStaking();
+        timelock.schedule(
+            address(deployment.stakingAdmin),
+            0,
+            abi.encodeWithSignature(
+                "upgradeAndCall(address,address,bytes)",
+                address(stakingProxy),
+                address(newStakingImplementation),
+                ""
+            ),
+            bytes32(0),
+            bytes32(0),
+            block.timestamp + 1 days
+        );
+        console.log("STAKING: upgrade scheduled");
+    }
+
+    function _deployStlMove() internal {
+        console.log("STL: deploying");
+        stlMoveToken stlMoveImplementation = new stlMoveToken();
+        vm.recordLogs();
+        stlMoveProxy = new TransparentUpgradeableProxy(
+            address(stlMoveImplementation),
+            address(timelock),
+            abi.encodeWithSignature(stlMoveSignature, "STL Move Token", "STL", address(moveProxy))
+        );
+        console.log("STL deployment records:");
+        console.log("proxy", address(stlMoveProxy));
+        deployment.stlMove = address(stlMoveProxy);
+        deployment.stlMoveAdmin = _storeAdminDeployment();
+    }
+
+    function _upgradeStlMove() internal {
+        console.log("STL: upgrading");
+        stlMoveToken newStlMoveImplementation = new stlMoveToken();
+        timelock.schedule(
+            address(deployment.stlMoveAdmin),
+            0,
+            abi.encodeWithSignature(
+                "upgradeAndCall(address,address,bytes)",
+                address(stlMoveProxy),
+                address(newStlMoveImplementation),
+                ""
+            ),
+            bytes32(0),
+            bytes32(0),
+            block.timestamp + 1 days
+        );
+        console.log("STL: upgrade scheduled");
+    }
+
+    function _deployMCR(address[] memory proposers) internal {
+        console.log("MCR: deploying");
+        MCR mcrImplementation = new MCR();
+        vm.recordLogs();
+        mcrProxy = new TransparentUpgradeableProxy(
+            address(mcrImplementation),
+            address(timelock),
+            abi.encodeWithSignature(
+                mcrSignature,
+                address(stakingProxy),
+                128,
+                100 ether,
+                100 ether, 
+                proposers
+            )
+        );
+        console.log("MCR deployment records:");
+        console.log("proxy", address(mcrProxy));
+        deployment.mcr = address(mcrProxy);
+        deployment.mcrAdmin = _storeAdminDeployment();
+    }
+
+    function _upgradeMCR() internal {
+        console.log("MCR: upgrading");
+        MCR newMCRImplementation = new MCR();
+        timelock.schedule(
+            address(deployment.mcrAdmin),
+            0,
```

Comment:
This just informs the hash of the operation. Right now, we would want these to collide to avoid redundant upgrades, right? Does that take care of everything? In what case would this take a nonce value?

Reply:
This actually schedules a transaction on the Timelock contract. It performs the call.
It's not feasible for us to include handling the execution of the timelock transaction imo. We'll be using the safe FE anyway, it takes way less time, it populates the transaction for us, handles signature for all the parties involved, etc.

Reply:
> safe FE

I don't recall agreeing on this. This supposed to be dispatched from CI.

Reply:
> > safe FE
> 
> I don't recall agreeing on this. This supposed to be dispatched from CI.

The 0 was for value in eth to be transactioned

Reply:
> The 0 was for value in eth to be transactioned

Wdym?

Reply:
during deployment you have the option of passing a certain amount of `eth` to subsequent calls because you might want to pay for something. This does not happen here.